chapter_10_external_events.asciidoc

```diff
@@ -55,22 +55,224 @@ events|in                                              events|out
+
+Person(customer, "Customer", "Wants to buy furniture")
+
+System(orders, "Orders", "Knows about customer orders")
+System(warehouse, "Warehouse", "Knows about shipping instructions")
+
+Rel(customer, orders, "Places order with")
+Rel(orders, batches, "Reserves stock with")
+Rel(batches, warehouse, "Sends instructions to")
+
+@enduml
+----   
+
+Each "thing" in our system has an associated service, which exposes an HTTP API.
+Our users visit a website and can choose from products that are in stock. When
+they add an item to their basket, we will reserve some stock for them. When an
+order is complete, we confirm the reservation, which causes us to send dispatch
+instructions to the warehouse.
+If this is the customer's third order, we want to update the customer record to
+flag them as a VIP.
+
+[role="image-source"]
+----
+[plantuml, chapter_10_sequence_1]
+@startuml
+
+actor Customer
+boundary Warehouse
+entity Orders
+entity Batches
+database CRM
+
+
+== Reservation ==
+
+  Customer -> Orders: Add product to basket
+  Orders -> Batches: Reserve stock
+  Batches --> Orders: Confirmed
+  Orders --> Customer: OK
+
+== Purchase ==
+
+  Customer -> Orders: Place order
+  Orders -> Batches: Confirm reservation
+  Batches -> Warehouse: Dispatch goods
+  Warehouse --> Batches: Dispatch received
+  Batches --> Orders: Reservation confirmed
+  Orders -> CRM: Update customer record
+  CRM --> Orders: Record updated
+  Orders --> Customer: Thanks for your order!
+
+
+@enduml
+----
+
+We can think of each of these steps as a Command in our system: Reserve Stock,
+Confirm Reservation, Dispatch Goods, Make Customer VIP.
+
+This style of architecture, where we create a microservice per database table,
+and treat our HTTP APIs as CRUD interfaces to anemic models is the most common
+way for people to approach service-oriented design.
+
+This works _fine_ for systems that are very simple, but quickly degrades into
+a Distributed Ball of Mud.
+
+To see why, let's consider another case. Sometimes, when stock arrives at the
+warehouse, we discover that items have been water damaged during transit. We
+can't sell water damaged sofas, so we have to throw them away, and request more
+stock from our partners. We also need to update our stock model, and that
+might mean we need to re-allocate a customer's order.
+
+Where does this logic go?
+
+Well, the warehouse system knows that the stock has been damaged, so maybe it
+should own this process?
+
+[role="image-source"]
+----
+[plantuml, chapter_10_sequence_2]
+@startuml
+
+actor w as "Warehouse worker"
+boundary Warehouse
+entity Orders
+entity Batches
+database CRM
+
+
+  w -> Warehouse: Stock damaged
+  Warehouse -> Batches: Decrease available stock
+  Batches -> Batches: Reallocate orders
+  Batches -> Website: Allocation changed
+  Orders -> CRM: Update order
+  CRM --> Website: Updated
+  Website --> Batches: Order history updated
+  Batches --> Warehouse: Stock decreased
+
+@enduml
+----
+
```

Comment:
What triggers the `Orders -> CRM: Update order` command/event to happen here?  (I tried rendering this in PlantUML, so it may just be an artifact of the tool).